### ISSUE TITLE
Example UseDraggable implementation does not work

### ISSUE THREAD

**Issue by GNUGradyn** [mgcrea/react-native-dnd#6]
Hello. I implemented UseDraggable as documented. The events are firing but the element is not following the cursor. I have made an MRE here. 

https://github.com/GNUGradyn/RnDndMre

Using the provided Draggable component works, but not the custom one in this repo based on the sample on the docs

**Comment by BrianJVarley** [comment]
@GNUGradyn I'm really looking forward to start using & added this package to a `react-native@0.72.3` project. I can see the `onBegin` - `onDragEnd` events firing in my console. But like you explained the `DraggableComponent` doesn't actually move or translate onto the `Droppable` component. I'm wondering if the latest version of  this package is not yet compatible with latest `react-native@0.72.3`? @mgcrea

**Comment by mgcrea** [comment]
@GNUGradyn in your example your custom component is lacking the required animatedStyle to actually move it along the drag gesture, you need something like this:

```tsx
import { useDraggable } from "@mgcrea/react-native-dnd";
import { Text, View } from "react-native";
import Animated, {
  useAnimatedStyle,
  withSpring,
} from "react-native-reanimated";

const DraggableComponent = ({ id, data, disabled }) => {
  const { offset, setNodeRef, activeId, setNodeLayout, draggableState } = useDraggable({
    id,
    data,
    disabled,
  });


  const animatedStyle = useAnimatedStyle(() => {
    const isActive = activeId.value === id;
    return {
      opacity: isActive ? 0.9 : 1,
      zIndex: isActive ? 999 : 1,
      transform: [
        {
          translateX: isActive ? offset.x.value : withSpring(offset.x.value),
        },
        {
          translateY: isActive ? offset.y.value : withSpring(offset.y.value),
        },
      ],
    };
  }, [id]);

  return (
    <Animated.View ref={setNodeRef} onLayout={setNodeLayout} style={animatedStyle}>
      <View style={{width: 50, height: 50, backgroundColor:"green"}}>
      <Text>DRAG</Text>
      </View>
    </Animated.View>
  );
};

export default DraggableComponent;
```

**Comment by mgcrea** [comment]
I did encounter issues when bundling the library for past release and did not immediately notice it on my local test setup, for more info see https://github.com/software-mansion/react-native-reanimated/discussions/4979

For now you must use the exact same reanimated version that the one from the lib (currently the latest) or you will encounter version mismatch issues, can you retry your code with `@mgcrea/react-native-dnd@1.5.5` and `react-native-reanimated@3.4.2`? It should work. Let me know if it does not and I'll investigate	further.

**Comment by BrianJVarley** [comment]
@mgcrea , yeah that's great this works now with `@mgcrea/react-native-dnd@1.5.6` based on your example 👏

**Comment by BrianJVarley** [comment]
@mgcrea This is what happens with your latest demo code, the Draggable component snaps back to its starting coordinates (0:0) on the View. I've also played around with a custom Draggable component, which uses an `animatedStyle` - https://github.com/mgcrea/react-native-dnd/issues/6#issuecomment-1688940396. I can see that the [updater function](https://docs.swmansion.com/react-native-reanimated/docs/core/useAnimatedStyle/#updater) is transforming the component along the `translateX` & `translateY` coordinates.  But on dragEnd (`draggableState.value = 5`) its resetting to starting coordinates -  `offset.x.value = 0, offfset.y.value = 0`

![rn-dnd-end-drag-behaviour](https://github.com/mgcrea/react-native-dnd/assets/6731968/3db37af8-ec46-4750-83df-f36cdeedf92a)



For example in the logs below you can see coordinates when I drag over the drop component, then let go of the DraggableComponennt, which resets to x & y = 0. Instead of dropping at the last coordinate before I let go of the item (offset.x.value = 0 & offset.y.value = -0.013319007820699134)

In the meantime I'll study the reanimated docs to better understand these animations.
But any help on this would be greatly appreciated.

```Bash
 LOG  draggableState.value = 5
 LOG  isActive = false
 LOG  activeId.value = null
 LOG  id = bend-1-key
 LOG  offset.x.value = 0
 LOG  offset.y.value = -0.013319007820699134
 LOG  data.over = {"over":{"id":"drop","data":{"value":{},"_value":{},"_animation":null,"_isReanimatedSharedValue":true},"disabled":false}}
 LOG  draggableState.value = 5
 LOG  isActive = false
 LOG  activeId.value = null
 LOG  id = bend-1-key
 LOG  offset.x.value = 0
 LOG  offset.y.value = 0
 LOG  data.over = {"over":{"id":"drop","data":{"value":{},"_value":{},"_animation":null,"_isReanimatedSharedValue":true},"disabled":false}}
```

**Comment by BrianJVarley** [comment]
I see my mistake after reading https://docs.swmansion.com/react-native-reanimated/docs/core/useAnimatedStyle#returns, 
just need to return an empty object when draggableState = `END` 

Although once I drop an item, I can no longer move it again. So it seems there is still a flaw in the animated style below?


```
  const animatedStyle = useAnimatedStyle(() => {
    const isActive = activeId.value === id;

    if (isActive && draggableState.value !== 5) {
      return {
        opacity: isActive ? 0.9 : 1,
        zIndex: isActive ? 999 : 1,
        transform: [
          {
            translateX: isActive ? offset.x.value : withSpring(offset.x.value),
          },
          {
            translateY: isActive ? offset.y.value : withSpring(offset.y.value),
          },
        ],
      };
    } else {
      return {};
    }
  }, [id, offset, activeId, draggableState]);
```

**Comment by mgcrea** [comment]
I've fixed reanimated bundling issues with the `1.6.0` release and everything should correctly works from now on, whatever the final app setup/dependencies is/are.

Feel free to open new issues if you encounter other issues.

@BrianJVarley your problem seems unrelated to this ticket so please open a separate one if you stil encounter issues.
